contracts/src/token/erc721/extensions/consecutive.rs

```diff
@@ -0,0 +1,1007 @@
+    ///
+    /// Overrides to ownership logic should be done to
+    /// [`Self::_owner_of`].
+    ///
+    /// # Errors
+    ///
+    /// If token does not exist, then the error
+    /// [`erc721::Error::NonexistentToken`] is returned.
+    ///
+    /// # Arguments
+    ///
+    /// * `&self` - Read access to the contract's state.
+    /// * `token_id` - Token id as a number.
+    pub fn _require_owned(&self, token_id: U256) -> Result<Address, Error> {
+        let owner = self._owner_of(token_id);
+        if owner.is_zero() {
+            return Err(erc721::Error::NonexistentToken(
+                ERC721NonexistentToken { token_id },
+            )
+            .into());
+        }
+        Ok(owner)
+    }
+}
+
+#[cfg(all(test, feature = "std"))]
+mod tests {
+    use alloy_primitives::{address, uint, Address, U256};
+    use stylus_sdk::msg;
+
+    use crate::{
+        token::{
+            erc721,
+            erc721::{
+                extensions::consecutive::{
+                    ERC721ExceededMaxBatchMint, Erc721Consecutive, Error,
+                },
+                tests::random_token_id,
+                ERC721InvalidReceiver, ERC721NonexistentToken, IErc721,
+            },
+        },
+        utils::structs::checkpoints::U96,
+    };
+
+    const BOB: Address = address!("F4EaCDAbEf3c8f1EdE91b6f2A6840bc2E4DD3526");
+
+    fn init(
+        contract: &mut Erc721Consecutive,
+        receivers: Vec<Address>,
+        batches: Vec<U96>,
+    ) -> Vec<U96> {
+        contract._first_consecutive_id.set(uint!(0_U96));
+        contract._max_batch_size.set(uint!(5000_U96));
+        receivers
+            .into_iter()
+            .zip(batches)
+            .map(|(to, batch_size)| {
+                contract
+                    ._mint_consecutive(to, batch_size)
+                    .expect("should mint consecutively")
+            })
+            .collect()
+    }
+
+    #[motsu::test]
+    fn mints(contract: Erc721Consecutive) {
+        let alice = msg::sender();
+
+        let initial_balance = contract
+            .balance_of(alice)
+            .expect("should return the balance of Alice");
+
+        let init_tokens_count = uint!(10_U96);
+        init(contract, vec![alice], vec![init_tokens_count]);
+
+        let balance1 = contract
+            .balance_of(alice)
+            .expect("should return the balance of Alice");
+        assert_eq!(balance1, initial_balance + U256::from(init_tokens_count));
+
+        // Check non-consecutive mint.
+        let token_id = random_token_id();
+        contract._mint(alice, token_id).expect("should mint a token for Alice");
+        let owner = contract
+            .owner_of(token_id)
+            .expect("should return the owner of the token");
+        assert_eq!(owner, alice);
+
+        let balance2 = contract
+            .balance_of(alice)
+            .expect("should return the balance of Alice");
+
+        assert_eq!(balance2, balance1 + uint!(1_U256));
```

Comment:
I'm usually leaning towards `assert_eq!(<current_value>, <expected>)`. But may be not all the time. I guess same as you in [pausable contract](https://github.com/OpenZeppelin/rust-contracts-stylus/blob/4fb347f4c20d9990f29c6dc9b6351038060b2ece/contracts/src/utils/pausable.rs#L143). I think we can vote (if it will make sense) later which formatting is preferable and change it everywhere at once.